addons/escoria-core/game/core-scripts/esc_item.gd

```diff
@@ -642,51 +673,91 @@ func set_angle(deg: int, wait: float = 0.0):
 # - deg: The angle degree to set
 # - float Wait this amount of seconds until continuing with turning around
 func turn_to(object: Node, wait: float = 0.0):
-	_movable.turn_to(object, wait)
+	if is_movable:
+		_movable.turn_to(object, wait)
+	else:
+		escoria.logger.warn(
+			self,
+			"Node %s cannot use \"turn_to\". Its \"is_movable\" parameter is false." %self
+		)
 
 
+# Check everything is in place to play talk animations
+func check_talk_possible():
+	if is_movable and (_movable.last_dir < 0 \
+			or _movable.last_dir >= animations.speaks.size()):
+		escoria.logger.warn(
+			self,
+			"Node %s cannot talk. Its \"last_dir\" parameter is invalid : %s." \
+			%[self, _movable.last_dir]
+		)
+		return false
+	if animations == null:
+		escoria.logger.warn(
+			self,
+			"Node %s cannot talk. Its \"animations\" parameter is empty : %s." \
+			%self
+		)
+		return false
+	if animations.speaks.size() == 0:
+		escoria.logger.warn(
+			self,
+			"Node %s cannot talk. Its \"animations.speaks\" array is empty : %s." \
+			%self
+		)
+		return false
+	if not get_animation_player():
```

Comment:
Above you're using `!` instead of `not`.  I don't have a preference either way, but let's try to be consistent with what we use. Please pick whichever one you want to use and update the other usages of it for this PR. Thanks!

Reply:
The only one I could spot was this one "if (!is_exit or dont_apply_terrain_scaling)" which I hadn't touched from the previous codebase. I've updated it to be "not" for consistency.

Reply:
Indeed, we historically chose to use `not` instead of `!`. I agree this is not the common usage in other languages though, but that's the choice we made :)